server/src/test/java/io/deephaven/server/session/SessionStateTest.java

```diff
@@ -88,6 +88,15 @@ public void teardown() {
         executionContext.close();
     }
 
+    @Test
+    public void testSessionIdInitialized() {
+        if (TestExecutionContext.SESSION_ID == null) {
+            Assert.neqNull(session.getSessionId(), "sessionId");
+        } else {
+            Assert.notEquals(TestExecutionContext.SESSION_ID, "sessionId", session.getSessionId());
+        }
+    }
```

Comment:
This seems like a bad test... ideally, you would have an explicit test case that triggered each of these code paths independently.